lib/xml-reader/src/main/java/org/triplea/generic/xml/reader/annotations/XmlOffset.java

```diff
@@ -0,0 +1,24 @@
+package org.triplea.generic.xml.reader.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Add this annotation to java fields to indicate their value should be mapped to the offset of the
+ * element in the XML file. This can be useful when a parent element supports different types of
+ * child elements and you need to know their relative order.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * class Tag {
+ *   @XmlOffset
+ *   private int xmlOffset;
```

Comment:
I don't quite understand the need for the 'xmlOffset', what is meant by an offset?
I wonder if we need this, so I'm curious about what it is used for.

Reply:
The way our xml parsing code works is it puts all child elements of a given tag into a collection.

So if you have:

```xml
<foo>
<bar/>
<bar/>
<baz/>
<bar/>
<baz/>
</foo>
```

You just get two collections, one of `bar` and one of `baz` and you can't tell their relative order. By using this, each of those elements keeps track of its offset in the XML, so we can reason about their relative order, which is needed here since the list of steps can have a mixture of `step` tags and `forEach` tags and their order is relevant.

Reply:
If order is important, it can become an attribute of a tag, eg:
```
<foo>
 <bar order="1"/>
 <baz order="2"/>
</foo>
```
In theory the order of XML tags generally should not matter (but often it does, and some XML specs rely on parsers to provide that guarantee).